src/toil/cwl/cwltoil.py

```diff
@@ -1805,6 +1806,77 @@ def determine_load_listing(tool: ToilCommandLineTool):
         )
     return load_listing
 
+class NoAvailableJobStoreException(Exception):
+    """Indicates that no job store name is available."""
+    pass
+
+def generate_default_job_store(batch_system_name: Optional[str], provisioner_name: Optional[str],
+                               local_directory: str) -> str:
+    """
+    Choose a default job store appropriate to the requested batch system and
+    porvisioner, and installed modules. Raises an error if no good default is
+    available and the user must choose manually.
+
+    :param batch_system_name: Registry name of the batch system the user has
+           requested, if any.
+    :param provisioner_name: Name of the provisioner the user has requested,
+           if any, May be 'aws' or 'gce'.
```

Comment:
Type hinting is `Optional[str]`, should this include `None`?

Reply:
Yes, I will note that None is allowed.